### PR TITLE
Added Tutor in CI

### PR DESCRIPTION
**Pull request by Anas12091101**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4443

### Description (What does it do?)
This PR replaces devstack with tutor in the CI for initializing edx-platform and running the integration tests of plugins. More details in the [ticket](https://github.com/mitodl/hq/issues/4443)

### How can this be tested?
- Please see the CI job logs
- To test locally, try to follow the job steps from the CI on your local machine.( this will take a while to run )
   - Build all the plugins packages using `pants build ::`
   - install tutor:
       - Nightly for Master Branch
          ```
           git clone --branch=nightly https://github.com/overhangio/tutor.git
           pip install -e "./tutor"
           ```
       - For Quince branch `pip install tutor==17.0.6`
       - For Redwood branch `pip install tutor==18.0.0`
   - Go to edx-platform directory and run `tutor mounts add .`
   - Build tutor images by running `tutor images build openedx-dev`
   - Run the launch command of tutor to complete the setup `tutor dev launch --non-interactive`
   - Stop the running containers `tutor dev stop`
   - Run the following commands:
       ```
        DEV="tutor_dev"  #"tutor_nightly_dev" in case of master branch
        docker-compose -f $(tutor config printroot)/env/local/docker-compose.yml -f $(tutor config printroot)/env/dev/docker-compose.yml --project-name $DEV run -v $PWD/../open-edx-plugins:/open-edx-plugins lms /open-edx-plugins/run_devstack_integration_tests.sh       
       ```
   - Validate the tests running and passing successfully

### CODE REVIEW

**Comment by Anas12091101** [comment]
This PR is waiting for the redwood release. For now, it is only working for quince release. It's failing on master because the master branch of edx-platform is broken due to a dependency version issue. For redwood, the tutor is not yet ready (see the release notes for the latest major release [v17.0.0](https://github.com/overhangio/tutor/releases/tag/v17.0.0) for tutor).

File: run_devstack_integration_tests.sh [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 mkdir -p test_root  # for edx
 paver update_assets lms --settings=test_static_optimized
 
 cp test_root/staticfiles/lms/webpack-stats.json test_root/staticfiles/webpack-stats.json
-
 cd /open-edx-plugins
 
-# Installing dev dependencies
-pip install poetry
-poetry install --no-interaction --only dev
```
Comment (Anas12091101) [inline run_devstack_integration_tests.sh:21]: Removed this code of installing dev dependencies as this resulted in a conflict with tutor edx-platform dependencies.

File: pyproject.toml [hunk 2]
```diff
@@ -22,15 +22,6 @@ opentelemetry-instrumentation-django = "*"
...
 
-[tool.poetry.group.dev.dependencies]
-black = "*"
-isort = "*"
-pre-commit = "^3.0.0"
-pytest = "^8.0.0"
-twine = "^5.0.0"
-ruff = "^0.4.0"
-pytest-mock = "^3.14.0"
-
```
Comment (blarghmatey) [inline pyproject.toml:33]: Why did this block get deleted? We can legitimately get rid of black and isort, but the rest should be retained.

**Review by blarghmatey — COMMENTED** [review]
Aside from the deletion of the dev dependencies, the rest of the changes here look good.

File: pyproject.toml (hunk 2 above)
Comment (Anas12091101) [reply pyproject.toml:33]: I thought we were only using these dependencies in the CI, therefore deleted them, will re-add them in the next commit.

**Review by blarghmatey — APPROVED** [review]

**Pull request merged**